handler/src/test/java/io/netty/handler/ssl/SslHandlerTest.java

```diff
@@ -63,4 +65,34 @@ public void testNonByteBufNotPassThrough() throws Exception {
 
         ch.writeOutbound(new Object());
     }
+
+    private class ReadSnooper extends ChannelOutboundHandlerAdapter {
+      public volatile boolean readIssued;
+      @Override
+      public void read(ChannelHandlerContext ctx) throws Exception {
+        readIssued = true;
+        super.read(ctx);
+      }
+    }
+
+    @Test
+    public void testIssueReadAfterActiveWriteFlush() throws Exception {
+      SSLEngine engine = SSLContext.getDefault().createSSLEngine();
+      engine.setUseClientMode(true);
+
+      ReadSnooper snooper = new ReadSnooper();
+      EmbeddedChannel ch = new EmbeddedChannel(); // issues channel active
+      ch.pipeline().removeLast(); // removes LastInboundsHandler that terminates the pipeline
+
+      ch.config().setAutoRead(false);
+      assertFalse(ch.config().isAutoRead());
+
+      ch.pipeline().addFirst(snooper);
+      // SslHandle will start handshaking on handlerAdded
+      ch.pipeline().addLast(new SslHandler(engine));
```

Comment:
The SslContext build via SslContextBuilder can then be used to create the SslHandler